### PR TITLE
Adding legacy error handling for co/auth endpoint

### PR DESCRIPTION
**Pull request by luisrudge**

### CODE REVIEW

File: test/web-auth/cross-origin-authentication.test.js [hunk 1]
```diff
@@ -166,7 +166,9 @@ describe('auth0.WebAuth.crossOriginAuthentication', function() {
               error_description: 'a huge error string'
             },
             code: 'any error',
```
Comment (joshcanhelp) [inline test/web-auth/cross-origin-authentication.test.js]: This should probably be a machine_name without spaces

File: test/web-auth/cross-origin-authentication.test.js [hunk 2]
```diff
@@ -166,7 +166,9 @@ describe('auth0.WebAuth.crossOriginAuthentication', function() {
               error_description: 'a huge error string'
             },
             code: 'any error',
-            description: 'a huge error string'
+            description: 'a huge error string',
+            error: 'any error',
```
Comment (joshcanhelp) [inline test/web-auth/cross-origin-authentication.test.js]: This should probably be a machine_name without spaces

File: src/helper/response-handler.js [hunk 3]
```diff
@@ -42,7 +42,10 @@ function wrapCallback(cb, options) {
       if (err.err) {
         err = err.err;
       }
-
+      if (options.forceLegacyError) {
+        errObj.error = err.error;
```
Comment (joshcanhelp) [inline src/helper/response-handler.js]: This means that it's possible for `errObj.code` !== `errObj.error` ... is that intended? Maybe this should fall after where `err.code` and `errObj.description` are set and set them as equal?

File: test/web-auth/cross-origin-authentication.test.js (hunk 2 above)
Comment (luisrudge) [reply test/web-auth/cross-origin-authentication.test.js]: doesn't really matter for the tests though..

File: src/helper/response-handler.js (hunk 3 above)
Comment (luisrudge) [reply src/helper/response-handler.js]: makes sense. pushed a change

**Pull request merged**